packages/Discretization/src/DTK_PointSearch_decl.hpp

```diff
@@ -0,0 +1,222 @@
+        Kokkos::View<double **, DeviceType> filtered_points,
+        Kokkos::View<int *, DeviceType> filtered_query_ids,
+        Kokkos::View<int *, DeviceType> filtered_ranks );
+
+    /**
+     * Keep data corresponding to points found inside the reference cell.
+     *
+     * @note This function should be <b>private</b> but lambda functions can
+     * only be called from a public function in CUDA.
+     */
+    void filterInCell(
+        std::array<Kokkos::View<bool *, DeviceType>, DTK_N_TOPO> const
+            &filtered_per_topo_point_in_cell,
+        std::array<Kokkos::View<double **, DeviceType>, DTK_N_TOPO> const
+            &filtered_per_topo_reference_points,
+        std::array<Kokkos::View<int *, DeviceType>, DTK_N_TOPO> const
+            &filtered_per_topo_cell_indices,
+        std::array<Kokkos::View<int *, DeviceType>, DTK_N_TOPO> const
+            &filtered_per_topo_query_ids,
+        std::array<Kokkos::View<int *, DeviceType>, DTK_N_TOPO> const &ranks,
+        std::array<Kokkos::View<int *, DeviceType>, DTK_N_TOPO>
+            &filtered_ranks );
+
+  private:
+    /**
+     * Compute the number of cells associated to each topology.
+     */
+    std::array<unsigned int, DTK_N_TOPO> computeNCellsPerTopology(
+        Kokkos::View<DTK_CellTopology *, DeviceType> cell_topologies );
+
+    /**
+     * Compute the position in the reference frame of candidates found by the
+     * search.
+     */
+    void performPointInCell(
+        Kokkos::View<double ***, DeviceType> cells,
+        Kokkos::View<unsigned int **, DeviceType> bounding_box_to_cell,
+        Kokkos::View<int *, DeviceType> imported_cell_indices,
+        Kokkos::View<Point *, DeviceType> imported_points,
+        Kokkos::View<int *, DeviceType> imported_query_ids,
+        Kokkos::View<int *, DeviceType> imported_ranks,
+        Kokkos::View<unsigned int *, DeviceType> topo, unsigned int topo_id,
+        Kokkos::View<double **, DeviceType> filtered_points,
+        Kokkos::View<int *, DeviceType> filtered_cell_indices,
+        Kokkos::View<int *, DeviceType> filtered_query_ids,
+        Kokkos::View<double **, DeviceType> reference_points,
+        Kokkos::View<bool *, DeviceType> point_in_cell,
+        Kokkos::View<int *, DeviceType> ranks );
+
+    /**
+     * Build the target-to-source distributor.
+     */
+    void build_distributor( std::array<Kokkos::View<int *, DeviceType>,
+                                       DTK_N_TOPO> const &filtered_ranks );
+
+    Teuchos::RCP<const Teuchos::Comm<int>> _comm;
+    Tpetra::Distributor _target_to_source_distributor;
+    unsigned int _dim;
+    std::array<Kokkos::View<Coordinate **, DeviceType>, DTK_N_TOPO>
+        _reference_points;
+    std::array<Kokkos::View<int *, DeviceType>, DTK_N_TOPO> _query_ids;
+    std::array<Kokkos::View<int *, DeviceType>, DTK_N_TOPO> _cell_indices;
+    std::array<Kokkos::View<double ***, DeviceType>, DTK_N_TOPO> _block_cells;
```

Comment:
`_block_cells` does not seem to be used in `getSearchResults()`.  Are you planning on using it in your interpolation operator?

Reply:
c.f. #369 ;)